src/array/cuda/uvm/array_index_select_uvm.cu

```diff
@@ -48,8 +49,13 @@ NDArray IndexSelectCPUFromGPU(NDArray array, IdArray index) {
           block.y *= 2;
       }
       const dim3 grid((len+block.y-1)/block.y);
-      CUDA_KERNEL_CALL(IndexSelectMultiKernel, grid, block, 0, thr_entry->stream,
-          array_data, num_feat, idx_data, len, ret_data);
+      if (num_feat * sizeof(DType) < 2 * CACHE_LINE_SIZE) {
```

Comment:
If the data is not aligned we will use the dgl's IndexSelectMultiKernel, otherwise we will use IndexSelectMultiKernelAligned which is specialized for UnifiedTensor?

Reply:
We use `IndexSelectMultiKernelAligned` for the unaligned case. The if else condition there is for the case where the node feature size is too small. If the node feature size is too small (e.g. only 1-2 GPU cacheline size), we don't have enough number of memory requests generated per feature access in the first place to attempt to coalesce the PCIe requests.

In summary:
`IndexSelectMultiKernel` -> The existing indexing kernel in DGL. This does not have the alignment fix.
`IndexSelectMultiKernelAligned` -> An UnifiedTensor specific indexing kernel which applies the alignment fix. Has a little value for the small node features and thus in such case simply fall back to `IndexSelectMultiKernel`.